processor/resourcedetectionprocessor/internal/resourcedetection.go

```diff
@@ -134,17 +136,41 @@ func (p *ResourceProvider) detectResource(ctx context.Context) {
 
 	p.logger.Info("began detecting resource information")
 
-	for _, detector := range p.detectors {
-		r, schemaURL, err := detector.Detect(ctx)
-		if err != nil {
-			p.logger.Warn("failed to detect resource", zap.Error(err))
-			if allowErrorPropagationFeatureGate.IsEnabled() {
-				p.detectedResource.err = err
+	resultsChan := make([]chan resourceResult, len(p.detectors))
+	for i, detector := range p.detectors {
+		resultsChan[i] = make(chan resourceResult)
+		go func(detector Detector) {
+			sleep := 1 * time.Second
+			for {
+				r, schemaURL, err := detector.Detect(ctx)
+				if err == nil {
+					resultsChan[i] <- resourceResult{resource: r, schemaURL: schemaURL, err: nil}
+					return
+				}
+				p.logger.Warn("failed to detect resource", zap.Error(err))
+				if allowErrorPropagationFeatureGate.IsEnabled() {
```

Comment:
@dashpole not sure how the behavior should be when the feature gate is enabled, since in the current state in this case there is no retry logic (which is probably something what we do not want). Any thoughts on this?